### ISSUE TITLE
ModalViewContainer should not monopolize Dialog.setOnKeyListener

### ISSUE THREAD

**Issue by rjrjr** [square/workflow-kotlin#259]
We use `Dialog.setOnKeyListener` for back button handling. If a subclass tries to set its own listener, it'll get stomped. Instead we should wrap `Dialog.window.callBack`.

**Comment by rjrjr** [comment]
e.g.

```kotlin
   buildDialog(view)
      .let { dialog ->
        val window = dialog.window!!
        // This is probably the dialog itself, but let's get it from the window
        // just in case someone else is also getting clever and wrapping things.
        val realWindowCallback = window.callback

        object : Window.Callback by realWindowCallback {
          override fun dispatchKeyEvent(event: KeyEvent): Boolean {
            return when {
              event.keyCode == KeyEvent.KEYCODE_BACK && event.action == ACTION_UP -> {
                initialViewEnvironment[RootViewBackPressedHandler].onBackPressed(view)
                true
              }
              else -> realWindowCallback.dispatchKeyEvent(event)
            }
          }
        }.also { myWindowCallback -> window.callback = myWindowCallback }

//...
```

**Comment by rjrjr** [comment]
Fixed by #595
